### PR TITLE
OCPBUGSM-30350: Add Progress info to Agent Status

### PR DESCRIPTION
**Pull request by rollandf**
# Description
Add Progress field in Agent Status that includes:
 - CurrentStage
 - ProgressInfo
 - StageStartTime
 - StageUpdateTime

Signed-off-by: Fred Rolland <frolland@redhat.com>

# What environments does this code impact?

- [ ] Automation (CI, tools, etc)
- [ ] Cloud
- [x] Operator Managed Deployments
- [ ] None

# How was this code tested?

Please, select one or more if needed:

- [x] assisted-test-infra environment
- [ ] dev-scripts environment
- [ ] Reviewer's test appreciated
- [x] Waiting for CI to do a full test run
- [ ] Manual (Elaborate on how it was tested)
- [ ] No tests needed

It's recommended to take a few extra minutes to provide more information about
how this code was tested. Here are some questions that may be worth answering:

- Should this PR be tested by the reviewer?
- Is this PR relying on CI for an e2e test run?
- Should this PR be tested in a specific environment?
- Any logs, screenshots, etc that can help with the review process?

Test-infra Agent CR Status snippet:
```
  progress:
    currentStage: Rebooting
    stageStartTime: "2021-06-15T06:45:32.877Z"
    stageUpdateTime: "2021-06-15T06:45:32.877Z"
```

# Assignees

Please, add one or two reviewers that could help review this PR.

/assign @filanov 
/assign @nmagnezi 
/assign @danielerez 

## Checklist

- [x] Title and description added to both, commit and PR.
- [x] Relevant issues have been associated (see [CONTRIBUTING] guide)
- [x] Reviewers have been listed
- [x] This change does not require a documentation update (docstring, `docs`, README, etc)
- [x] Does this change include unit-tests (note that code changes require unit-tests)

## Reviewers Checklist

- [ ] Are the title and description (in both PR and commit) meaningful and clear?
- [ ] Is there a bug required (and linked) for this change?
- [ ] Should this PR be backported?

[Kubernetes community documentation]: https://github.com/kubernetes/community/blob/master/contributors/guide/pull-requests.md#commit-message-guidelines
[CONTRIBUTING]: https://github.com/openshift/assisted-service/blob/master/CONTRIBUTING.md

### CODE REVIEW

**Review by nmagnezi — APPROVED** [review]
/lgtm

**Comment by flaper87** [comment]
@rollandf will this be backported? Technically, this changes the CRD API.

It might be better to get this in now in 4.8 rather than keeping it just for 4.9 and then introducing a breaking change. What do you think?

File: internal/controller/api/v1beta1/agent_types.go [hunk 1]
```diff
@@ -125,9 +125,9 @@ type HostProgressInfo struct {
 	CurrentStage models.HostStage `json:"currentStage,omitempty"`
 	ProgressInfo string           `json:"progressInfo,omitempty"`
 	// Name in REST API: stage_started_at
-	StageStartTime string `json:"stageStartTime,omitempty"`
+	StageStartTime metav1.Time `json:"stageStartTime,omitempty"`
```
Comment (nmagnezi) [inline internal/controller/api/v1beta1/agent_types.go:128]: should this be `*metav1.Time` ?

File: internal/controller/api/v1beta1/agent_types.go [hunk 2]
```diff
@@ -125,9 +125,9 @@ type HostProgressInfo struct {
...
 	// Name in REST API: stage_updated_at
-	StageUpdateTime string `json:"stageUpdateTime,omitempty"`
+	StageUpdateTime metav1.Time `json:"stageUpdateTime,omitempty"`
```
Comment (nmagnezi) [inline internal/controller/api/v1beta1/agent_types.go:130]: ditto

**Review by nmagnezi — CHANGES_REQUESTED** [review]
/lgtm

**Comment by rollandf** [comment]
> @rollandf will this be backported? Technically, this changes the CRD API.
> 
> It might be better to get this in now in 4.8 rather than keeping it just for 4.9 and then introducing a breaking change. What do you think?

Yes, it will be backported

**Review by nmagnezi — APPROVED** [review]
/lgtm

File: internal/controller/api/v1beta1/agent_types.go (hunk 1 above)
Comment (rollandf) [reply internal/controller/api/v1beta1/agent_types.go]: Done

File: internal/controller/api/v1beta1/agent_types.go (hunk 2 above)
Comment (rollandf) [reply internal/controller/api/v1beta1/agent_types.go]: Done

**Pull request merged**